File: lms/lms/page/sample/sample.js
// frappe.pages[''].on_page_load = function(wrapper) {
// 	var page = frappe.ui.make_app_page({
// 		parent: wrapper,
// 		title: 'LMS',
// 		single_column: true
		
// 	});
	
// }
frappe.pages['sample'].on_page_load = function(wrapper) {
	var page = frappe.ui.make_app_page({
	  parent: wrapper,
	  title: 'LMS',
	  single_column: true
	});
  
	// Create a custom button
	wrapper.page.add_inner_button(__('Get Time Table'), function() {
	  // Get the current user's standard
	  frappe.call({
		method: 'lms.lms.doctype.sample.sample.get_details',
		args: {
		 
		 'user':frappe.session.user
		},
		callback: function(response) {
		  var userStandard = response.message.standard;
  
		  // Navigate to the Time Table doctype with filters
		  frappe.set_route('List', 'Time Table', { standard: userStandard });
		}
	  });
	});
  };
  

// frappe.pages['sample'].on_page_load = function(wrapper) {
// 	// Fetch the user profile details using the frappe.request method
// 	frappe.request({
// 	  method: 'GET',
// 	  url: '/api/resource/User/' + frappe.session.user,
// 	  callback: function(response) {
// 		console.log(response);
// 		var user_profile = response;
// 		console.log(user_profile);
// 		// Display the user profile details in the page
// 		if (user_profile) {
// 		  wrapper.page.main.find('.your-profile-element').text(user_profile.name1);
// 		  // Add any other profile details you want to display
// 		}
// 	  }
// 	});
//   };

File: lms/lms/page/sample/sample.py
import frappe
# def get_context(context):
#     session_user = frappe.session.user

#     # Retrieve additional user details
#     user_details = frappe.get_doc("User", session_user)

#     # Pass the user details to the context
#     context.session_user = session_user
#     context.full_name = user_details.full_name
#     context.email = user_details.email
#     # Add any other user details you requ
#   # Create a custom button
#   context.custom_buttons = [
#     {
#       'label': 'Custom Button',
#       'onclick': 'your_custom_function()',
#       'icon': 'fa fa-star',
#       'style': 'color: red;'
#     }
#   ]
#    return context



def get_context(context):
    # Fetch the user profile details
    user_profile = frappe.get_doc("User", frappe.session.user).as_dict()
    print("user_profile".user_profile)

    # Pass the user profile details to the context
    context.user_profile = user_profile
    # Add any other profile details you require

    # Render the page with the updated context
    return context

